src/Engine/ProtoScript/Runners/LiveRunner.cs

```diff
@@ -1721,6 +1728,14 @@ private void SynchronizeInternal(GraphSyncData syncData)
                 var guids = runtimeCore.ExecutedAstGuids.ToList();
                 executedAstGuids[syncData.SessionID] = guids;
                 runtimeCore.RemoveExecutedAstGuids();
+
+                // There should be a CodeBlock in CodeBlockList by now
+                if (runnerCore.CodeBlockList.Any())
+                {
+                    var nodes = runnerCore.CodeBlockList[(int)Language.Associative].instrStream.dependencyGraph.GetGraphNodesAtScope(Constants.kInvalidPC, Constants.kInvalidPC);
+                    //delete all nodes which were redefined. For those nodes that were defined for the first time, this will be a no-op
+                    nodes.RemoveAll(x=>!x.isActive || pendingUIDsForDeletion.Contains(x.guid));
```

Comment:
Is this enough ? Isn't it possible that the children or parents of each node contain references to the nodes we remove ?
So should we also remove pendingUIDsForDeletion from the children and parents of each node ?

Reply:
As for now, Children nodes are fully recalculated in each run in `BuildGraphNodeDependencies` according to which variables they depend on so it shouldn't be necessary to delete dependant / parent nodes.

But currently this code will keep modified nodes until the final deletion comes, so in a sense thid deletion is incomplete - i just haven't find a way to perfectly remove only the necessary nodes yet and am erring in the side of caution.